pkg/maps/eppolicymap/eppolicymap.go

```diff
@@ -36,7 +36,7 @@ var (
 
 const (
 	// MaxEntries represents the maximum number of endpoints in the map
-	MaxEntries = 65535
+	MaxEntries = 65536
```

Comment:
Any implications during the upgrade due to the bump?

Reply:
Great question. Let me explore this a bit..

Existing `sockmap` programs will continue to reference the old `eppolicymap` and each entry in the old map holds a reference to the map that the `eppolicymap` points to. So lookups coming in before endpoints are regenerated will continue to work, even when the map is recreated/pinned on the filesystem.

So what happens next? Well, we start regenerating endpoints. For each endpoint that we regenerate, it triggers update of the new `eppolicymap` map with the endpoint's `policymap` fd so that socket programs will be able to reference that endpoint.

I'm not sure if there is any ordering between the (re)load of socket programs vs. all endpoints regeneration, but let's say that these happen at the same time. When the socket progs are loaded, they may reference the new map, which doesn't have links to all local endpoint `policymap` fds. This means that sockmap progs will not be able to find the policy (including proxy redirects) from arbitrary endpoints. However the only code in the `bpf/sockops/bpf_sockops.c` that relies upon this is the code which updates the `sock_ops_map` to support redirects to proxy sockets; During upgrade, proxy sockets are disrupted anyway because we redeploy the entire proxy.

Furthermore, sockmap is not yet out of beta so I don't know if upgrade considerations are paramount yet.

In conclusion, no, there should be no implications during upgrade.

Reply:
Also if we fail the lookup in the ENDPOINTS_MAP then we will just pass the connection onto the stack and worse case sockops will not be used.

Reply:
Ah right, I missed the redirect program. But yes, that will just fall back to the packet path (and the BPF enforcement that occurs there).

Reply:
I think there's actually still a bit more subtlety here where this would effectively turn off sockmap redirection for existing connections during upgrade. This could have some follow-on effects but it's kinda hard to say for sure how it will interact.

I guess this suggests that we should consider setting up a variation on our upgrade tests where we run with sockmap ;)